File: yaf-gadomain/src/main/java/ar/edu/ungs/yamiko/ga/domain/impl/BasicGenotype.java
package ar.edu.ungs.yamiko.ga.domain.impl;

import java.io.Serializable;
import java.util.List;

import ar.edu.ungs.yamiko.ga.domain.Chromosome;
import ar.edu.ungs.yamiko.ga.domain.Genotype;

/**
 * Implementación Básica de Genotipo.
 * @author ricardo
 *
 * @param <T>
 */
public class BasicGenotype<T> implements Genotype<T>,Serializable{

	/**
	 * 
	 */
	private static final long serialVersionUID = -6048609693910215627L;
	private List<Chromosome<T>> chromosomes;
	
	public BasicGenotype(List<Chromosome<T>> _chromosomes) {
		chromosomes=_chromosomes;
	}
	
	public List<Chromosome<T>> getChromosomes() {
		return chromosomes;
	}
	
	
}

File: yaf-gadomain-scala/src/main/scala/ar/edu/ungs/yamiko/ga/domain/impl/BasicGenotype.scala
package ar.edu.ungs.yamiko.ga.domain.impl

import ar.edu.ungs.yamiko.ga.domain.Genotype
import ar.edu.ungs.yamiko.ga.domain.Chromosome

@SerialVersionUID(1399L)
class BasicGenotype[T](chromosomes:List[Chromosome[T]] ) extends Genotype[T]{
  
  override def getChromosomes():List[Chromosome[T]]=chromosomes

  def canEqual(a: Any) = a.isInstanceOf[BasicGenotype[T]]

  override def equals(that: Any): Boolean =
    that match {
      case that: BasicGenotype[T] => 
        that.canEqual(this) && this.hashCode == that.hashCode
      case _ => false
  }

  override def hashCode:Int = {
    val ourHash = if (chromosomes == null) 0 else chromosomes.hashCode 
    super.hashCode + ourHash      
  }
   
}